### PR TITLE
Add flags to use client certs for cli.

### PR DESCRIPTION
**Pull request by Capstan**
This allows `grpc_cli` to act with the specific client identity when using SSL. It does _not_ however set the cert when using Google Default Credentials, as it does not have the necessary options to provide a client cert and key.

### CODE REVIEW

**Comment by Capstan** [comment]
Partially fixes #16059.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by Capstan** [comment]
If not @yang-g, then perhaps @yashykt can review?

File: test/cpp/util/cli_credentials.cc [hunk 1]
```diff
@@ -64,7 +77,27 @@ CliCredentials::GetChannelCredentials() const {
...
+      grpc_slice_unref(cert_slice);
+    }
+    if (!FLAGS_ssl_client_key.empty()) {
+      grpc_slice key_slice = grpc_empty_slice();
+      GRPC_LOG_IF_ERROR(
+          "load_file",
+          grpc_load_file(FLAGS_ssl_client_key.c_str(), 1, &key_slice));
+      ssl_creds_options.pem_private_key =
+          grpc::StringFromCopiedSlice(key_slice);
+      grpc_slice_unref(key_slice);
```
Comment (yashykt) [inline test/cpp/util/cli_credentials.cc]: use grpc_slice_unref_internal

File: test/cpp/util/cli_credentials.cc [hunk 2]
```diff
@@ -64,7 +77,27 @@ CliCredentials::GetChannelCredentials() const {
...
+    grpc::SslCredentialsOptions ssl_creds_options;
+    // TODO(@Capstan): This won't affect Google Default Credentials using SSL.
+    if (!FLAGS_ssl_client_cert.empty()) {
+      grpc_slice cert_slice = grpc_empty_slice();
+      GRPC_LOG_IF_ERROR(
+          "load_file",
+          grpc_load_file(FLAGS_ssl_client_cert.c_str(), 1, &cert_slice));
+      ssl_creds_options.pem_cert_chain =
+          grpc::StringFromCopiedSlice(cert_slice);
+      grpc_slice_unref(cert_slice);
```
Comment (yashykt) [inline test/cpp/util/cli_credentials.cc]: use grpc_slice_unref_internal

File: test/cpp/util/cli_credentials.cc (hunk 1 above)
Comment (Capstan) [reply test/cpp/util/cli_credentials.cc]: Done.

File: test/cpp/util/cli_credentials.cc (hunk 2 above)
Comment (Capstan) [reply test/cpp/util/cli_credentials.cc]: Done.

**Comment by Capstan** [comment]
Mergeable is still failing because it needs release notes and/or language labels.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by yashykt** [comment]
Adding @yang-g as second reviewer

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,952,970      Total (=)      1,952,970

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,665,044      Total (<)     10,665,045

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,952,970      Total (=)      1,952,970

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,665,042      Total (=)     10,665,042

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by Capstan** [comment]
Ping. @yang-g

File: test/cpp/util/cli_credentials.cc (hunk 2 above)
Comment (yang-g) [reply test/cpp/util/cli_credentials.cc]: @yashykt Why should the internal version be used here?

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,956,097      Total (=)      1,956,097

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,882      Total (<)     10,843,883

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,956,097      Total (=)      1,956,097

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,887      Total (>)     10,843,882

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: test/cpp/util/cli_credentials.cc (hunk 2 above)
Comment (Capstan) [reply test/cpp/util/cli_credentials.cc]: Reverted and repushed.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,960,882      Total (=)      1,960,882

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,881,534      Total (=)     10,881,534

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**